test/AdaptiveSpinLockTest.cpp

```diff
@@ -62,27 +62,39 @@ TEST(AdaptiveSpinLockTest, WorksWithStdUniqueLock) {
   lock.unlock();
 }
 
-const int thread_num = 7;
-int val;
-pipert::AdaptiveSpinLock* g_lock = nullptr;
+namespace {
 
-void SharedIncrementor(int times) {
-  ASSERT_TRUE(g_lock);
-  for (int i = 0; i < times; i++) {
-    std::lock_guard<pipert::AdaptiveSpinLock> guard(*g_lock);
-    val++;
+class SharedIncrementStore {
+ public:
+  SharedIncrementStore(const int threads) : threads_(threads){};
+
+ public:
+  void Increment(int cycles) {
+    std::lock_guard<pipert::AdaptiveSpinLock> guard(slock_);
+    for (int i = 0; i < cycles; ++i) val_++;
```

Comment:
This locks the mutex and does all the modifications all at once. This is very efficient, but in the original test, the idea was to try testing the thing by locking and unlocking the mutex for every modification. That is testing really what would happen if multiple threads would always reace for the same resource.

Reply:
I also have an idea here. We could simulate a very different scenarion by not only locking and unlocking for a single modification, but we could also de a yield() after every succesful unlock. This way, the OS scheduler would be more likely mix threads to have a potential clash for resources. This could be a separate, modified test.